lib/ash_authentication/strategies/magic_link/request.ex

```diff
@@ -35,7 +35,12 @@ defmodule AshAuthentication.Strategy.MagicLink.Request do
              {sender, send_opts} <- strategy.sender,
              {:ok, token} <-
                MagicLink.request_token_for_identity(strategy, identity, context_opts, context) do
-          sender.send(to_string(identity), token, Keyword.put(send_opts, :tenant, context.tenant))
+          build_opts =
+            send_opts
+            |> Keyword.put(:tenant, context.tenant)
```

Comment:
Lets use `Keyword.merge` here.

Reply:
I mean instead of:

```elixir
 build_opts =
            send_opts
            |> Keyword.put(:tenant, context.tenant)
            |> Keyword.put(:source_context, context.source_context)
```

we should use 

```elixir
 build_opts = Keyword.merge(send_opts, tenant: context.tenant, source_context: context.source_context)
```